packages/@expo/cli/src/start/server/middleware/RuntimeRedirectMiddleware.ts

```diff
@@ -36,7 +37,7 @@ export class RuntimeRedirectMiddleware extends ExpoMiddleware {
   async handleRequestAsync(req: ServerRequest, res: ServerResponse): Promise<void> {
     const { query } = parse(req.url!, true);
     const isDevClient = query['choice'] === 'expo-dev-client';
-    const platform = parsePlatformHeader(req);
+    const platform = parsePlatformHeader(req) ?? resolvePlatformFromUserAgentHeader(req);
```

Comment:
`parsePlatformHeader` is used across almost all middleware, why should the policy for platform detection change for just one use-case.